vllm/model_executor/models/llava_onevision.py

```diff
@@ -824,13 +825,55 @@ def apply_pooling(self, image_features, stride=2):
         image_feature = image_feature.view(batch_frames, -1, dim)
         return image_feature
 
+    def get_multimodal_embeddings(
+            self, **kwargs) -> Optional[List[Tuple[NestedTensors, str]]]:
+        modalities = self._parse_and_validate_multimodal_inputs(**kwargs)
+        if not modalities:
+            return None
+
+        # We append modality representation to each embedding. This is a
+        # temporary workaround for models that can handle multiple modalities
+        # at the same time.
+        multimodal_embeddings: List[Tuple[NestedTensors, str]] = []
+
+        if "images" in modalities:
+            image_input = modalities["images"]
+            vision_embeddings = self._process_image_input(image_input)
+            multimodal_embeddings.append((vision_embeddings, "image"))
+        if "videos" in modalities:
+            video_input = modalities["videos"]
+            video_embeddings = self._process_video_pixels(video_input)
+            multimodal_embeddings.append((video_embeddings, "video"))
```

Comment:
I'm aware this is not clean. 

I'm thinking at model runner level we should be able to store the modality information, and `get_input_embeddings` will have an additional `modality` parameter. Open to suggestions

Reply:
Just so I understand correctly, is the proposal to have a class like
```python
class MultimodalEmbedding:
    placeholder_token_id: int
    data: NestedTensors
```
instead of `NestedTensors` for output of `get_multimodal_embeddings`?

Reply:
@Isotr0py I played around with this proposal, but it actually made things a bit more complicated. With `NestedTensors`, it's pretty straightforward for us to split, slice and combine back embeddings between encoder cache engine and model runner in V1.

Reply:
Hmmm, yea, if using wrapper, it might need to overload the operator for slicing etc, which could be duplicated...

I was just thinking about "ignore" the modality in encoder outputs caching and embeddings merge (the `get_input_embeddings`). Because there is not a very large difference to deal with different modality's embeddings in these steps.

Reply:
Per our offline discussion, we will leave the design for mixed-modalities for later.